timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py

```diff
@@ -308,8 +337,10 @@ def predict(self, data: TimeSeriesDataFrame, quantile_levels: List[float] = None
         # Make sure the item_ids are sorted in the same order as in data
         return df.loc[data.item_ids]
 
-    def _predict_gluonts_forecasts(self, data: TimeSeriesDataFrame, **kwargs) -> List[Forecast]:
-        gts_data = self._to_gluonts_dataset(data)
+    def _predict_gluonts_forecasts(
+        self, data: TimeSeriesDataFrame, known_covariates: Optional[TimeSeriesDataFrame] = None, **kwargs
+    ) -> List[Forecast]:
+        gts_data = self._to_gluonts_dataset(data, known_covariates=known_covariates)
```

Comment:
1. This shouldn't happen in the first place since we ensure that the correct columns are passed to each inside `TimeSeriesLearner.predict`
2. The underlying model will raise an exception if the columns are missing.

I still added an extra check / exception here to make sure that we get an interpretable error message and not some "can't concatenate" exception from the model's internals